### PR TITLE
Adding More Tests

### PR DESCRIPTION
**Pull request by mehradsadeghi**
- DRYing unit tests.
- Adding one more unit test.
- Adding integration tests for `database` and `eloquent` providers.

### CODE REVIEW

**Comment by imanghafoori1** [comment]
Good job

**Comment by imanghafoori1** [comment]
@mehradsadeghi 
You may consider my new package if you want to continue helping out with my repos:
https://github.com/imanghafoori1/eloquent-history

Thanks in advance

**Comment by mehradsadeghi** [comment]
> @mehradsadeghi
> You may consider my new package if you want to continue helping out with my repos:
> https://github.com/imanghafoori1/eloquent-history
> 
> Thanks in advance

@imanghafoori1 

Honestly when you announced (on LinkedIn) that you're going to do a new package in a way that "even though maintaining open source repos are a difficult responsibility, But I'm gonna do this package anyway", I wanted to respond to that post right away and declare my readiness to help.

But there were couple of challenges in the way :)
First of all I have a full time job which takes my time and energy more than normal sometimes.
Also I wanted to dive into more advanced packages of yours (like Microscope) and learn from it or even contribute to it (considering my limited free time and energy).

So long story short, I didn't want to make a promise and break it, So I didn't say anything on Linkedin :))

But I feel honored that you considered me for contribution to `eloquent-history`, So I try my best to do something about it.
